features/flows/classification/engine/impl/src/main/java/org/opennms/netmgt/flows/classification/internal/DefaultClassificationService.java

```diff
@@ -116,7 +115,7 @@ public Integer saveRule(Rule rule) throws InvalidRuleException {
             // persist
             group.addRule(rule);
             final Integer ruleId = classificationRuleDao.save(rule);
-            updateRulePositionsAndReloadEngine(group);
+            updateRulePositionsAndReloadEngine(RulePrioritizationUtil.sortRulePositions(group));
```

Comment:
Can we make the sorting part of the `updateRulePositionsAndReloadEngine` method?